packages/flutter_tools/lib/src/flutter_manifest/parse_result.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+sealed class ParseResult<T> {
+  const ParseResult();
+
+  List<String> get errors;
+}
+
+final class ValueParseResult<T> extends ParseResult<T> {
+  const ValueParseResult(this.value);
+  final T value;
+
+  // Defining `errors` here allows callers only interested in validation (and not
+  // getting the parsed value) to simply call `errors` without having to `switch`
```

Comment:
if you don't need to switch, why do we even need a sum type? could this not be a single class, with a value and error field?